app/controlplane/plugins/core/guac/v1/guac.go

```diff
@@ -0,0 +1,268 @@
+	return nil
+}
+
+// uploadToBucket uploads the provided content to the bucket under a deterministic, unique name (digest + filename)
+// It also sets the content type and additional metadata
+func uploadToBucket(ctx context.Context, bucket *storage.BucketHandle, fileName string, content []byte, md *sdk.ChainloopMetadata, logger *log.Helper) error {
+	bucketInfo, err := bucket.Attrs(ctx)
+	if err != nil {
+		return fmt.Errorf("getting bucket info: %w", err)
+	}
+
+	buf := bytes.NewBuffer(content)
+	fileSize := uint64(buf.Len())
+	logger.Infow("msg", "writing to the bucket", "file", fileName, "bucket", bucketInfo.Name, "size", bytefmt.ByteSize(fileSize))
+
+	w := bucket.Object(fileName).NewWriter(ctx)
+	defer w.Close()
+
+	// Set metadata and content type
+	// application/json can't be detected automatically by https://mimesniff.spec.whatwg.org/#supplied-mime-type-detection-algorithm
+	// If not set, the underlying library will try to sniff it automatically
+	if filepath.Ext(fileName) == ".json" {
+		w.ObjectAttrs.ContentType = "application/json"
+	}
+
+	w.ObjectAttrs.Metadata = map[string]string{
+		"author":          "chainloop",
+		"workflowID":      md.WorkflowID,
+		"workflowName":    md.WorkflowName,
+		"workflowProject": md.WorkflowProject,
+		"workflowRunID":   md.WorkflowRunID,
+		"filename":        fileName,
+	}
+
+	if _, err := io.Copy(w, buf); err != nil {
+		return fmt.Errorf("writing to the bucket: %w", err)
+	}
+
+	logger.Infow("msg", "uploaded to the bucket", "file", fileName, "bucket", bucketInfo.Name, "size", bytefmt.ByteSize(fileSize))
+
+	return nil
+}
+
+// loadBucket returns a bucket handle if:
+// 1. The credentials are valid
+// 2. The bucket exists
+// 3. The credentials have write permissions
+func loadBucket(ctx context.Context, client *storage.Client, bucket string) (*storage.BucketHandle, error) {
+	if bucket == "" {
+		return nil, errors.New("no bucket provided")
+	}
+
+	// Check that the bucket exists
+	b := client.Bucket(bucket)
+	_, err := b.Attrs(ctx)
+	if err != nil {
+		if errors.Is(err, storage.ErrBucketNotExist) {
+			return nil, fmt.Errorf("the bucket %s does not exist", bucket)
+		}
+
+		return nil, fmt.Errorf("checking the bucket: %w", err)
+	}
+
+	// Write test file to make sure credentials have write permissions
+	testFile := b.Object("chainloop-test-write")
+	if err := testFile.NewWriter(ctx).Close(); err != nil {
+		return nil, fmt.Errorf("we can't write to the bucket %s: %w", bucket, err)
+	}
+	_ = testFile.Delete(ctx)
+
+	return b, nil
+}
+
+// validateExecuteOpts validates that we receive all the required options, from materials to credentials
+func validateExecuteOpts(m *sdk.ExecuteMaterial, regConfig *sdk.RegistrationResponse, _ *sdk.AttachmentResponse) error {
+	if m == nil || m.Content == nil {
+		return errors.New("invalid input")
+	}
+
+	if m.Type != schemaapi.CraftingSchema_Material_SBOM_CYCLONEDX_JSON.String() && m.Type != schemaapi.CraftingSchema_Material_SBOM_SPDX_JSON.String() {
```

Comment:
Can we somehow automatically generate this validation? We are already providing the list of accepted material types above.

Reply:
That's a good point, I don't like it either. As you said, this validation is kind of pointless. The execute method will contain only once of those material types.

I am going to remove this code.